src/shared/Microsoft.Git.CredentialManager/Interop/Linux/LinuxCredentialStore.cs

```diff
@@ -45,12 +49,62 @@ public bool Remove(string service, string account)
 
         private void EnsureBackingStore()
         {
-            if (_backingStore is null)
+            if (_backingStore != null)
             {
-                // TODO: determine the available backing stores based on the current session
-                // TODO: prompt for the desired backing store
-                // TODO: store the desired backing store to ~/.gitconfig
-                _backingStore = new SecretServiceCollection(_namespace);
+                return;
+            }
+
+            string ns = _settings.CredentialNamespace;
+            string credStoreName = _settings.CredentialBackingStore?.ToLowerInvariant();
+
+            switch (credStoreName)
+            {
+                case "secretservice":
+                    ValidateSecretService();
+                    _backingStore = new SecretServiceCollection(ns);
+                    break;
+
+                case "plaintext":
```

Comment:
I'm a bit surprised these string literals are not used anywhere else (except the documentation below...). I wonder if it is worth placing them in a constants class?

Reply:
I can move them into a constant file, yes.